src/std_extensions/arithmetic/int_types.rs

```diff
@@ -244,6 +251,40 @@ mod test {
             ConstIntS::new(3, 128),
             Err(ConstTypeError::CustomCheckFail(_))
         );
-        assert_matches!(ConstIntS::new(3, -128), Ok(_));
+        assert!(ConstIntS::new(3, -128).is_ok());
+
+        let const_u32_7 = const_u32_7.unwrap();
+        assert!(const_u32_7.equal_consts(&ConstIntU::new(5, 7).unwrap()));
+        assert_eq!(const_u32_7.log_width(), 5);
+        assert_eq!(const_u32_7.value(), 7);
+        assert!(const_u32_7
+            .check_custom_type(&int_custom_type(TypeArg::BoundedNat { n: 5 }))
+            .is_ok());
+        assert!(const_u32_7
+            .check_custom_type(&int_custom_type(TypeArg::BoundedNat { n: 6 }))
+            .is_err());
+
+        assert_eq!(const_u32_7.name(), "u5(7)");
+        assert!(const_u32_7
+            .check_custom_type(&int_custom_type(TypeArg::BoundedNat { n: 19 }))
+            .is_err());
+
+        let const_i32_2 = ConstIntS::new(5, -2).unwrap();
+        assert!(const_i32_2.equal_consts(&ConstIntS::new(5, -2).unwrap()));
+        assert_eq!(const_i32_2.log_width(), 5);
+        assert_eq!(const_i32_2.value(), -2);
+        assert!(const_i32_2
+            .check_custom_type(&int_custom_type(TypeArg::BoundedNat { n: 5 }))
+            .is_ok());
+        assert!(const_i32_2
+            .check_custom_type(&int_custom_type(TypeArg::BoundedNat { n: 6 }))
+            .is_err());
+        assert!(const_i32_2
+            .check_custom_type(&int_custom_type(TypeArg::BoundedNat { n: 19 }))
+            .is_err());
+        assert_eq!(const_i32_2.name(), "i5(-2)");
+
+        ConstIntS::new(50, -2).unwrap_err();
```

Comment:
You could combine these two into a single test `new(50, 0)` - 0 is legal for *any* integer type...

Reply:
not sure what you mean here - I'd still need two lines for the unsigned and signed

Reply:
I had to dive into `fn new` to understand what this was about but after that I thought the point was that it was gonna ignore the second argument

Reply:
it's testing that both constants error on invalid widths (coverage)

Reply:
Yes, so they both cover exactly the same code, as the "is width valid" check is done before even looking at the second argument??

Reply:
One covers 90-94, the other covers 120-124 ?

Reply:
Ok, my bad. Somehow I'd seen +2/-2 but not S/U, sorry!